From 5cf6afbd36721761ee17b87d906e9b17753c0881 Mon Sep 17 00:00:00 2001
From: Levente Polyak <levente@leventepolyak.net>
Date: Sun, 13 Jan 2019 21:42:45 +0100
Subject: [PATCH 094/102] Revert "mark kernel_set_to_readonly as
 __ro_after_init"

This reverts commit 296db046b8688b96332d54e273cac265644a1e47.

CPA conflicts detected when marking kernel_set_to_readonly as
__ro_after_init #4
---
 arch/x86/mm/init_32.c | 5 +++--
 arch/x86/mm/init_64.c | 5 +++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/arch/x86/mm/init_32.c b/arch/x86/mm/init_32.c
index 0cfe04b639af..963c729c163b 100644
--- a/arch/x86/mm/init_32.c
+++ b/arch/x86/mm/init_32.c
@@ -837,7 +837,7 @@ void arch_remove_memory(int nid, u64 start, u64 size,
 }
 #endif
 
-int kernel_set_to_readonly __ro_after_init;
+int kernel_set_to_readonly __read_mostly;
 
 static void mark_nxdata_nx(void)
 {
@@ -861,11 +861,12 @@ void mark_rodata_ro(void)
 	unsigned long start = PFN_ALIGN(_text);
 	unsigned long size = (unsigned long)__end_rodata - start;
 
-	kernel_set_to_readonly = 1;
 	set_pages_ro(virt_to_page(start), size >> PAGE_SHIFT);
 	pr_info("Write protecting kernel text and read-only data: %luk\n",
 		size >> 10);
 
+	kernel_set_to_readonly = 1;
+
 #ifdef CONFIG_CPA_DEBUG
 	pr_info("Testing CPA: Reverting %lx-%lx\n", start, start + size);
 	set_pages_rw(virt_to_page(start), size >> PAGE_SHIFT);
diff --git a/arch/x86/mm/init_64.c b/arch/x86/mm/init_64.c
index 4c8dfc0d7ca0..9c1520a170fd 100644
--- a/arch/x86/mm/init_64.c
+++ b/arch/x86/mm/init_64.c
@@ -1255,7 +1255,7 @@ void __init mem_init(void)
 	mem_init_print_info(NULL);
 }
 
-int kernel_set_to_readonly __ro_after_init;
+int kernel_set_to_readonly;
 
 void mark_rodata_ro(void)
 {
@@ -1268,9 +1268,10 @@ void mark_rodata_ro(void)
 
 	printk(KERN_INFO "Write protecting the kernel read-only data: %luk\n",
 	       (end - start) >> 10);
-	kernel_set_to_readonly = 1;
 	set_memory_ro(start, (end - start) >> PAGE_SHIFT);
 
+	kernel_set_to_readonly = 1;
+
 	/*
 	 * The rodata/data/bss/brk section (but not the kernel text!)
 	 * should also be not-executable.
-- 
2.28.0

